src/source_map_utils.ts

```diff
@@ -0,0 +1,72 @@
+import {SourceMapConsumer, SourceMapGenerator} from 'source-map';
+
+const INLINE_SOURCE_MAP_REGEX =
+    new RegExp('^//# sourceMappingURL=data:application/json;base64,(.*)$', 'm');
+
+export function containsInlineSourceMap(source: string): boolean {
+  return getInlineSourceMapCount(source) > 0;
+}
+
+export function getInlineSourceMapCount(source: string): number {
+  const match = source.match(INLINE_SOURCE_MAP_REGEX);
+  if (match) {
+    // The INLINE_SOURCE_MAP_REGEX has a capture group, but we only want to count real matches
+    return match.length / 2;
+  } else {
+    return 0;
+  }
+}
+
+export function extractInlineSourceMap(source: string): string {
+  const result = INLINE_SOURCE_MAP_REGEX.exec(source)!;
```

Comment:
we've hard issues before where source code would contain:

```javascript

// #sourceMapURL...
more().code();
// #sourceMap...
```

I.e., you should take the last source map, not the first.